### PR TITLE
Making version of randcat for uniform diffusion, adding utils.jl, and…

### PR DESCRIPTION
**Pull request by murrellb**
… tidying

### CODE REVIEW

**Comment by murrellb** [comment]
@bicycle1885 Please check and merge!

**Comment by bicycle1885** [comment]
@murrellb I've added a lot of changes and cleanups. Can you check if you are happy with them?

File: src/rotational.jl [hunk 1]
```diff
@@ -1,17 +1,13 @@
 export
     RotDiffusionProcess,
-    MultiRotationState,
-    features
+    MultiRotationState
```
Comment (murrellb) [inline src/rotational.jl]: Do we need to export this? The user shouldn't see this type.

File: src/rotational.jl [hunk 2]
```diff
@@ -68,13 +64,14 @@ endpoint_conditioned_sample(rng::AbstractRNG, process::RotDiffusionProcess, s::R
 
 values(r::MultiRotationState) = copy(r.rots)
```
Comment (murrellb) [inline src/rotational.jl]: I don't think we need this anymore? It was used in the old tracking system, but now the rotations themselves should be returned by the tracker anyway.

File: src/rotational.jl (hunk 1 above)
Comment (bicycle1885) [reply src/rotational.jl]: I don't think we need to. I'll remove it.

File: src/rotational.jl (hunk 2 above)
Comment (bicycle1885) [reply src/rotational.jl]: Agreed. I'll drop MultiRotationState and its functions.

**Pull request merged**